fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -2784,6 +2798,14 @@ private void addIndexStateReadConflict(@Nonnull String indexName) {
         if (!getRecordMetaData().hasIndex(indexName)) {
             throw new MetaDataException("Index " + indexName + " does not exist in meta-data.");
         }
+        if (indexStateReadConflicts == null) {
+            indexStateReadConflicts = ConcurrentHashMap.newKeySet(8);
```

Comment:
Is there a reason this doesn't just set it to a `ConcurrentHashMap` in the constructor? We'll need to access this whenever we access an index, which will be essentially most things that interact with the store, so it seems like just initializing this in the constructor would be simpler and not much more expensive?

Reply:
Sure, let me fix.